login-workflow/src/screens/subScreens/AcceptEula/AcceptEula.tsx

```diff
@@ -0,0 +1,166 @@
+ *
+ * @param slots used for each slot in `ContactSupport`
+ *
+ * @param slotProps applied to each slot
+ *
+ * @category Component
+ */
+
+export const AcceptEula: React.FC<AcceptEulaProps> = (props) => {
+    const { t } = useLanguageLocale();
+    const {
+        eulaAccepted,
+        eulaContent,
+        onEulaChanged,
+        loadEula,
+        htmlEula,
+        eulaError,
+        agreeTerms = t('blui:REGISTRATION.EULA.AGREE_TERMS'),
+        eulaControl,
+        termsAndConditionsStyles,
+        eulaContentStyles,
+        EulaContentProps,
+        loaderText = t('blui:REGISTRATION.EULA.LOADING'),
+        loaderStyles,
+        LoaderTextProps,
+        TermsAndConditionsProps,
+        classes = {},
+        slots = {},
+        slotProps = {},
+    } = props;
+
+    const defaultClasses = useUtilityClasses(props);
+    const TermsAndConditions = slots.termsAndConditions ?? FormControlLabel;
+    const eulaContentInternals = eulaContent ?? eulaError ?? loaderText;
+
+    useEffect(() => {
+        void loadEula();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return (
+        <>
+            {!htmlEula && (
+                <Typography
+                    sx={{ flex: '1 1 0px', overflow: 'auto', ...loaderStyles }}
+                    className={cx(defaultClasses.loaderText, classes.loaderText)}
+                    component={slots.loaderText}
+                    {...LoaderTextProps}
+                    {...slotProps.loaderText}
+                >
+                    {eulaContentInternals}
+                </Typography>
+            )}
+            {htmlEula && (
+                <Box
+                    sx={{ flex: '1 1 0px', overflow: 'auto', ...eulaContentStyles }}
+                    className={cx(defaultClasses.eulaContent, classes.eulaContent)}
+                    component={slots.eulaContent}
+                    dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(eulaContentInternals) }}
+                    {...EulaContentProps}
+                    {...slotProps.eulaContent}
+                ></Box>
+            )}
+            <TermsAndConditions
+                control={
+                    eulaControl ?? (
+                        <Checkbox
+                            color={'primary'}
+                            checked={eulaAccepted}
+                            disabled={!eulaContent}
+                            onChange={(event: React.ChangeEvent<HTMLInputElement>): void =>
+                                onEulaChanged(event.target.checked)
+                            }
+                        />
+                    )
+                }
+                label={agreeTerms}
+                sx={{ flex: '0 0 auto', mr: 0, mt: 2, ...termsAndConditionsStyles }}
+                className={cx(defaultClasses.termsAndConditions, classes.termsAndConditions)}
+                {...TermsAndConditionsProps}
+                {...slotProps.termsAndConditions}
```

Comment:
Let's use only the `slotProps` here and get rid of that `TermsAndConditionsProps`.